src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -268,16 +279,15 @@ static File makeChunkFile(File baseDir, int i)
   public static class Outer implements SmooshedWriter
   {
     private final int fileNum;
-    private final OutputStream out;
     private final int maxLength;
+    private final GatheringByteChannel channel;
 
-    private boolean open = true;
     private int currOffset = 0;
 
-    Outer(int fileNum, OutputStream out, int maxLength)
+    Outer(int fileNum, FileChannel channel, int maxLength)
```

Comment:
Yep, I wish that too, but it was like that before. I thought it's possible once but also wanted minimize the changes.